src/core/mainloop/connection.c

```diff
@@ -449,6 +449,20 @@ connection_new(int type, int socket_family)
   }
 }
 
+static void
+connection_close_and_invalidate_socket(connection_t *conn)
+{
+  if (connection_speaks_cells(conn)) {
+    or_connection_t *or_conn = TO_OR_CONN(conn);
+    tor_tls_free(or_conn->tls);
+    or_conn->tls = NULL;
```

Comment:
`tor_tls_free()` should set `or_conn->tls` to `NULL`.

Reply:
Right; the compiler should remove this redundancy for us.